classes/ticket_processor_class.rb

```diff
@@ -0,0 +1,64 @@
+require 'sendgrid-ruby'
+
+class EmailClient
+  include SendGrid
+
+  def send_email(from, to, subject, message)
+    from = Email.new(email: from)
+    to = Email.new(email: to)
+    content = Content.new(type: 'text/plain', value: message)
+    mail = Mail.new(from, subject, to, content)
+    sg = SendGrid::API.new(api_key: ENV.fetch("SENDGRID_API_KEY"))
+    response = sg.client.mail._('send').post(request_body: mail.to_json)
+    puts response.status_code
+    puts response.body
+    puts response.headers
+  end
```

Comment:
A lot of things happen in this one method - it's a signal that maybe it would be clearer to divide it into smaller parts.

Do you have an idea what parts would make sense to separate?

Reply:
The first thing I see that seems like a separate thing is handling the response. You could create a separate private method for this:

```ruby
private

def log_response(response)
  puts response.status_code
  puts response.body
  puts response.headers
end
```

Reply:
Another thing could be preparing (building a `mail` object) the mail before sending:

```ruby
private

def build_mail(from, to, subject, message)
  from = Email.new(email: from)
  to = Email.new(email: to)
  content = Content.new(type: 'text/plain', value: message)
  Mail.new(from, subject, to, content)
end
```